src/element.ts

```diff
@@ -9,22 +9,33 @@
 import { Vex } from './vex';
 import { Registry } from './registry';
 import { Flow } from './tables';
-
-export class Element {
-  static newID() {
-    return 'auto' + Element.ID++;
-  }
-
-  constructor({ type } = {}) {
+import { BoundingBox } from './boundingbox';
+import { Font } from './smufl';
+import { RenderContext, ElementStyle, ElementAttributes } from './types/common';
+
+export abstract class Element {
+  protected static ID: number = 1000;
+  protected context?: RenderContext;
+  protected rendered: boolean;
+  protected style?: ElementStyle;
+  private attrs: ElementAttributes;
+  protected boundingBox?: BoundingBox;
+  protected fontStack: Font[];
+  protected musicFont: Font;
+  protected registry?: Registry;
+
+  static newID(): string {
+    return `auto${Element.ID++}`;
+  }
+
+  constructor(type: string) {
```

Comment:
This should be `constructor({type: string} = {})` -- it's a destructured object param.

Reply:
Ok after some googling to get it to compile I found the way to declare that with TypeScript:
`constructor({ type }: { type?: string } = {})` not very obvious in the first instance :)